stub/src/main/java/io/grpc/stub/ClientCalls.java

```diff
@@ -122,22 +117,52 @@
   public static <ReqT, RespT> Iterator<RespT> blockingServerStreamingCall(
       ClientCall<ReqT, RespT> call, ReqT param) {
     BlockingResponseStream<RespT> result = new BlockingResponseStream<RespT>(call);
-    asyncServerStreamingCall(call, param, result.listener());
+    asyncUnaryRequestCall(call, param, result.listener());
```

Comment:
It seems it would make more sense to leave this as `asyncServerStreamingCall`. Yes, that method can be implemented with unary, but server streaming is the semantics that we want.